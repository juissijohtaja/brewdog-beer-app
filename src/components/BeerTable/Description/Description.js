import React, { useState } from 'react'
import Tooltip from '@material-ui/core/Tooltip'
import { makeStyles } from '@material-ui/core/styles'
import Box from '@material-ui/core/Box'
import ClickAwayListener from '@material-ui/core/ClickAwayListener'

const useStyles = makeStyles({
  textWrap: {
    whiteSpace: 'nowrap',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
  },
  sizeLarge: {
    width: 180,
  },
})

const Description = ({ description }) => {

  const classes = useStyles()
  const [open, setOpen] = useState(false)

  const handleTooltipOpen = () => {
    setOpen(true)
  }

  const handleTooltipClose = () => {
    setOpen(false)
  }

  return (
    <Box data-testid='description'>
      <ClickAwayListener onClickAway={handleTooltipClose}>
        <Tooltip
          arrow
          title={description}
          onClose={handleTooltipClose}
          open={open}
          disableFocusListener
          disableHoverListener
          disableTouchListener>
          <Box
            onClick={ open ? handleTooltipClose : handleTooltipOpen }
            className={`${classes.textWrap} ${classes.sizeLarge}`}
          >
            {description}
          </Box>
        </Tooltip>
      </ClickAwayListener>
    </Box>
  )
}

export default Description