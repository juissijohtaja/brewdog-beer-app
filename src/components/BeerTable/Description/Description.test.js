import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import initialStoreState from '../../../setupTests'
import Description from './Description'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'
import configureStore from 'redux-mock-store'

const mockStore = configureStore([])

describe('Description Component', () => {
  let store, component

  beforeEach(() => {
    store = mockStore(initialStoreState)
    store.dispatch = jest.fn()

    component = renderer.create(
      <Provider store={store}>
        <Description description='Lorem ipsum dolor sit amet' />
      </Provider>
    )
  })

  test('should render Lorem ipsum dolor sit amet', () => {
    expect(component.toJSON()).toMatchSnapshot()
  })
})