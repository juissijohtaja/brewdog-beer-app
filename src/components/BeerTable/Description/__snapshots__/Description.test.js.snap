// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description Component should render Lorem ipsum dolor sit amet 1`] = `
<div
  className="MuiBox-root MuiBox-root-3"
  data-testid="description"
>
  <div
    aria-describedby={null}
    className="MuiBox-root MuiBox-root-4 makeStyles-textWrap-1 makeStyles-sizeLarge-2"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    title={null}
  >
    Lorem ipsum dolor sit amet
  </div>
</div>
`;
